### DISCUSSION TITLE
Exporting `themeConfigs` does not let `config` to be exported.

### DISCUSSION THREAD

**Discussion by TheAFKGamer10** [facebook/docusaurus#9255] [Q&A] [answered]
When i change the `docusaurus.config.js` file and run the development server it works fine, and no errors occur. But when i add 
```
module.exports = {
  themeConfig: {
    colorMode: {
      defaultMode: 'dark',
      disableSwitch: false,
      respectPrefersColorScheme: false,
    },
  },
};
```
to the bottom of the file, the console sends the error
```
[ERROR] Error: "baseUrl" is required
"title" is required
"url" is required

    at validateConfig (C:\Users\gumba\docusourus\node_modules\@docusaurus\core\lib\server\configValidation.js:228:15)
    at loadSiteConfig (C:\Users\gumba\docusourus\node_modules\@docusaurus\core\lib\server\config.js:40:62)
    at async loadContext (C:\Users\gumba\docusourus\node_modules\@docusaurus\core\lib\server\index.js:31:63)
    at async load (C:\Users\gumba\docusourus\node_modules\@docusaurus\core\lib\server\index.js:74:21)
```
If i remove the themeConfig from the file it works fine again with no errors and logs
```
√ Client
  Compiled successfully in 93.28ms

client (webpack 5.88.2) compiled successfully 
```
I have tried searching the web for this issue but nothing seems to solve this. Code taken from [the docs](https://docusaurus.io/docs/api/themes/configuration#color-mode---dark-mode). Any support on this?

**Comment by Josh-Cena** [comment]
You need to export everything in a single object. Think about it: if you have `module.exports = ...` twice, then the second one surely overrides the first.

```js
module.exports = {
  // The other config
  themeConfig: {
    // The other themeConfig
    colorMode: {
      defaultMode: 'dark',
      disableSwitch: false,
      respectPrefersColorScheme: false,
    },
  },
};
```

**Reply by TheAFKGamer10** [reply]
I'm either not doing it correctly or something is not working, but when i add `config,` or even `config: {},` to where `// Here` is it still does not work.
```js
module.exports = {
// Here
  themeConfig: {
    colorMode: {
      defaultMode: 'dark',
      disableSwitch: true,
      respectPrefersColorScheme: true,
    },
  },
};
```
I was thinking putting the `config,` because that is how it was before, i was also thinking `config: {},` because that is how the themeConfig is.
Am I missing something, or do i just not know how to move code. I am trying to include the original config, nothing special.

**Reply by slorber** [reply]
Your config does not look like a valid config. It should look like this:

```js
module.exports = {
  title: 'My Site',
  tagline: 'Dinosaurs are cool',
  favicon: 'img/favicon.ico',
  url: 'https://your-docusaurus-test-site.com',
  baseUrl: '/',
  themeConfig: {
    colorMode: {
      defaultMode: 'dark',
      disableSwitch: true,
      respectPrefersColorScheme: true,
    },
  },
};
```

Nowhere in our docs it is written to do something like this:

```js 
module.exports = {
  config: {
  },
  themeConfig: {
  },
};
```

I'm not sure why you had this idea, if you read the docs carefully and start again from the initial template, you should be fine.

**Reply by TheAFKGamer10** [reply]
I think I have figure it out, if I put it below the line `/** @type {import('@docusaurus/preset-classic').ThemeConfig} */` everything else works and everything but the default color works. The disable switch works but the default color goes to white when I have it set to dark. Any ideas on why this is happening?
```
  themeConfig:
    /** @type {import('@docusaurus/preset-classic').ThemeConfig} */
    ({
      colorMode: {
        defaultMode: 'dark',
        disableSwitch: false,
        respectPrefersColorScheme: false,
      },
```

**Reply by slorber** [reply]
Please help us help you, because the way you ask question is time consuming for us:
- don't paste partial configs
- create a repro of your problem using docusaurus.new
- explain what is actual vs expected behavior ("works" "does not work" is not helpful, describe properly)

**Reply by TheAFKGamer10** [reply]
[Here is my config file](https://pastebin.com/egEPA60c) https://pastebin.com/egEPA60c Starts on line 51.
I made a docusaurus.new site and it also does not set the default theme when it is spoused to set the default theme as dark mode. https://ctlf3t-3000.csb.app/

**Reply by slorber** [reply]
I need to see the code associated to your CSB site.
Please create a repro where I can see the code and run it easily. 
I can't trust that the code I see in pastebin is the code you run in the sandbox: a human mistake is possible and I don't want to spend time debugging that.

**Reply by TheAFKGamer10** [reply]
Here is my current code https://github.com/TheAFKGamer10/docusourushelp

**Reply by slorber** [reply, marked as answer]
Isn't it dark by default for you?

https://stackblitz.com/~/github.com/TheAFKGamer10/docusourushelp

It is for me.

<img width="1913" alt="CleanShot 2023-08-24 at 19 55 52@2x" src="https://github.com/facebook/docusaurus/assets/749374/d9f60383-c2a3-449a-b2f3-f7ff7c97cb35">


Note if you interact with the theme toggle, the last theme you selected is persisted in localStorage so make sure to use a new clean browser session to test the default theme.

**Reply by TheAFKGamer10** [reply]
Ok, thanks. I think that was my problem, i had clicked the color mode and it stays in the other browsers too. As long as it starts in dark mode i don't care about me. Thanks for the help.
